feat(audit): add notes for each flagged flatpak permission

HastD opened the pull request:
If a flatpak app can acquire arbitrary permissions due to access to a DBus name that's known to allow sandbox escapes, add a note stating the bus name, not just that the app can acquire arbitrary permissions. (This was already shown in the "recommendations" section but not as a note directly under the app name.)

Example of what this looks like:
<img width="1291" height="102" alt="image" src="https://github.com/user-attachments/assets/e1de5329-bc78-4c20-ba43-b7a45e60226a" />

Also add a note for the permission to modify flatpak permissions, and slightly refactor the handling of notes and recommendations associated to flatpak permissions so that it's harder to accidentally omit notes like this.

HastD committed:
feat(audit): show dangerous bus access for flatpak apps

If a flatpak app can acquire arbitrary permissions due to access to a
DBus name that's known to allow sandbox escapes, add a note stating the
bus name, not just that the app can acquire arbitrary permissions. (This
was already shown in the "recommendations" section but not as a note
directly under the app name.)

alexvojproc commented:
I think it'd be nice if we could somehow combine it onto one line? Like:
> This can acquire arbitrary permissions via org.freedesktop.login1 on the session bus.

Because otherwise it might seem that "acquire arbitrary permissions" is a separate thing to the D-Bus access, if that makes sense?

HastD committed:
feat(audit): improve handling of notes on flatpak permissions

This ensures that adding a note, adding a recommendation, and changing
the status are all done together for each flagged permission. As part of
this, added a note that was missing for flatpaks that have permission to
modify flatpak permissions.

HastD commented:
@alexvojproc That's a nice idea but would be somewhat awkward to implement given the current design. If you have an idea for how to improve this further, that could be a different PR. For now, I made an additional change to handle notes and recommendations more systematically for flatpak permissions, which brought to my attention that another permission was missing an accompanying note.

RoyalOughtness committed:
Merge branch 'live' into flatpak-bus-notes
